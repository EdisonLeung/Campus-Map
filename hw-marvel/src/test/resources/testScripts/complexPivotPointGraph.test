# Test pathfinding on a complex graph where all nodes point to 1 "center" node and
# are all connect to each other via that "center" node. Main purpose of test is to
# test shortest pathfinding on a complex graph

# Load Graph
LoadGraph pivotPoint pivotPointGraph.csv

# Pathfinding tests
FindPath pivotPoint e b

FindPath pivotPoint e c

FindPath pivotPoint e d

FindPath pivotPoint a b

FindPath pivotPoint a d

FindPath pivotPoint c d

FindPath pivotPoint b d

FindPath pivotPoint b c