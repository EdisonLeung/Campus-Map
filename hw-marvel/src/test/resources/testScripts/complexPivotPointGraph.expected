# Test pathfinding on a complex graph where all nodes point to 1 "center" node and
# are all connect to each other via that "center" node. Main purpose of test is to
# test shortest pathfinding on a complex graph

# Load Graph
loaded graph pivotPoint

# Pathfinding tests
path from e to b:
e to b via BE

path from e to c:
e to c via CE

path from e to d:
e to d via DE

path from a to b:
a to e via AE
e to b via BE

path from a to d:
a to c via AC
c to d via CD

path from c to d:
c to d via CD

path from b to d:
b to e via BE
e to d via DE

path from b to c:
b to e via BE
e to c via CE